### PR TITLE
67: Installed memcached session storage

### PR DESCRIPTION
**Pull request by katiefenn**
- Installed connect-memcached
- Removed Mongo session storage
- Added switch to revert to memory session storage if memcached not configured

### CODE REVIEW

File: server/helpers/session.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+export default function sessionStorageConfig(config) {
+  let settings = { ...SESSION_STORAGE_DEFAULTS };
+
+  if(config.get('SESSION:SECRET')) {
+    // TODO: Log using configured session secret
+    settings.secret = config.get('SESSION:SECRET');
+  } else {
+    // TODO: Warn using development site session secret
+    settings.secret = require('crypto').randomBytes(64).toString('hex');
```
Comment (katiefenn) [inline server/helpers/session.js]: Considering changing this to flatly refuse to start if secret is not supplied instead of just generating a random secret. Comments appreciated.

File: test/unit/reducers/t_identity.js [hunk 2]
```diff
@@ -1,7 +1,6 @@
 import expect from 'expect';
 
 import { identity } from '../../../src/reducers/identity';
-import * as ActionTypes from '../../../src/actions/identity';
```
Comment (katiefenn) [inline test/unit/reducers/t_identity.js]: Unrelated linting fix.

File: test/unit/server/t_middleware.js [hunk 3]
```diff
@@ -1,4 +1,3 @@
-import expect from 'expect';
```
Comment (katiefenn) [inline test/unit/server/t_middleware.js]: Unrelated linting fix.

File: test/unit/reducers/t_identity.js (hunk 2 above)
Comment (bartaz) [inline test/unit/reducers/t_identity.js]: These were already fixed in master I believe.

File: test/unit/server/helpers/session.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    let config;
+    let mockConfig = {};
+
+    beforeEach(() => {
+      mockConfig.get = sinon.stub();
+      mockConfig.get.withArgs('SESSION:SECRET').returns('TEST');
+      config = sessionStorageConfig(mockConfig);
+    });
+
+    it('should note return a session store', () => {
```
Comment (bartaz) [inline test/unit/server/helpers/session.js]: Small typo: `s/note/not/`

File: test/unit/server/helpers/session.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    let config;
+    let mockConfig = {};
+
+    beforeEach(() => {
+      mockConfig.get = sinon.stub();
+      mockConfig.get.withArgs('SESSION:SECRET').returns(null);
+      config = sessionStorageConfig(mockConfig);
+    });
+
+    it('should note return a session store', () => {
```
Comment (bartaz) [inline test/unit/server/helpers/session.js]: Same small typo: `s/note/not/`

File: server/helpers/session.js (hunk 1 above)
Comment (earnubs) [inline server/helpers/session.js]: Per discussion on IRC, this has been done.

File: README.md [hunk 6]
```diff
@@ -2,34 +2,29 @@
...
 
-Install dependencies:
+## npm packages:
 
     $ npm install
 
-and run it:
+# Starting Development Environment:
 
     $ npm start
```
Comment (earnubs) [inline README.md]: Inlining the env var means it's in history, and I don't have to bother with export
```
SESSION_SECRET="foo" npm start
```
